doc/mgmt/SONiC_PIM_SM_HLD.md

```diff
@@ -0,0 +1,1036 @@
+| char[INTERFACE_NAMSIZ]      | iif_name   | Incoming interface name |
+| struct ip_addr | src        | Source address of the route |
+| struct ip_addr | grp        | Group address of the route  |
+| mfpm_route_status_t         | entry_status | Entry aliveness status | 
+
+struct mfpm_kat_info_t
+
+|   Data Type    | Field_name |  Description   |
+| -------------- | ---------- | -------------- |
+| unit32_t       | vrf_id     | VRF Id         |
+| unsigned int   | keep_alive_time | Keep alive timer config to be applied in NPU |
+
+union mfpm_data_msg_t
+
+|   Data Type    | Field_name |  Description   |
+| -------------- | ---------- | -------------- |
+|  mfpm_kat_info_t | kat_info | Keep alive timer config to be set |
+|  mfpm_route_info_t | route_info | Info to get Route status from MFPM client |
+
+### 3.2.1 **SAI specific data structure and callback changes**
+
+Following data structures will be included and exposed to SYNCD to receive IPMC entry AGE and DONTAGE data,
+
+/**
+ * @brief Notification data format received from SAI IPMC callback
+  * @count attr[attr_count]
+ */
+
+typedef struct _sai_ipmc_event_notification_data_t
+{
+   
+   /** Event type */
+   
+    sai_ipmc_event_t event;
+    
+     /** IPMC entry */
+     
+    sai_ipmc_entry_t ipmc_entry;
+ 
+} sai_ipmc_event_notification_data_t;
+
+**sai_ipmc_entry_t** - This is an existing SAI data structure.
+
+/**
+ * @brief IPMC event type
+ */
+
+typedef enum _sai_ipmc_event_t
+{
+   
+   /** IPMC entry aged */
+   
+    SAI_IPMC_EVENT_AGED,
+ 
+    /** IPMC entry don’t age */
+    
+    SAI_IPMC_EVENT_DONT_AGE,
+ 
+} sai_ipmc_event_t;
+ 
+/**
+ * @brief IPMC notifications
+  * @count data[count]
+  * @param[in] count Number of notifications
+ * @param[in] data Pointer to IPMC event notification data array
+ */
+ 
+typedef void (*sai_ipmc_event_notification_fn)(
+        _In_ uint32_t count,
+        _In_ const sai_ipmc_event_notification_data_t *data);
+
+## 3.3 DB Changes
+PIM ASM leverages all the existing DBs. There are few new DBs introduced in CONFIG and STATE DBs like PIM_GLOBALS_STATIC_RP & IPMC_MROUTE_AGE_TABLE for ASM specific functionality.
+
+### 3.3.1 CONFIG DB
+#### 3.3.1.1 PIM Global Configuration [Existing]
+```JSON
+
+Key                       : PIM_GLOBALS|vrf-name|address-family
+
+vrf-Name                  : 1-15 characters ; VRF name
+address-family            : enum {ipv4, ipv6}
+
+join-prune-interval       : Join prune interval in seconds. Range (60-600 seconds)
+keep-alive-timer          : Keep alive timer in seconds. Range (31-60000 seconds)
+ssm-ranges                : Configure Source-Specific-Multicast group range using IP Prefix-list
+ecmp-enabled              : To enable PIM ECMP (true/false)
+ecmp-rebalance-enabled    : To enable PIM ECMP rebalance (true/false). It can be enabled/disabled only when ECMP is enabled
+spt-threshold             : To disable or re-enable the SPT switchover. By default, SPT switchover is enabled in the system.
```

Comment:
CLI section has prefix-list for spt disabled group list? where is that configuration stored?

Reply:
Added the prefix-list as well as corrected the name for spt-action